activerecord/lib/active_record/connection_adapters/abstract/query_comment.rb

```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/module/attribute_accessors_per_thread"
+
+module ActiveRecord
+  module ConnectionAdapters
+    module QueryComment
+      extend ActiveSupport::Concern
+      included do
+        mattr_accessor :prepend_comment, default: false
+      end
+
+      module ClassMethods
```

Comment:
@eileencodes what do you think about this approach to prepending the execution methods so that all adapters are handled in one place?